src/plugins/auto/auto_schedule.cpp

```diff
@@ -125,105 +124,181 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
         std::list<DeviceInformation> validDevices =
             _autoSContext->_plugin->GetValidDevice(_autoSContext->_devicePriorities,
                                                    _loadContext[ACTUALDEVICE].networkPrecision);
+        // When the hint is ctput and there is only one device, the single-device logic is used
         if (validDevices.size() == 1) {
-            // When the hint is ctput and there is only one device, the single-device logic is used instead of
-            // the MULTI logic
-            // can not change _autoSContext->_performanceHint to THROUGHPUT, because GetMetric needs to return CTPUT
             _loadContext[ACTUALDEVICE].deviceInfo = validDevices.front();
             _loadContext[ACTUALDEVICE].deviceInfo.config[CONFIG_KEY(PERFORMANCE_HINT)] =
                 IE::PluginConfigParams::THROUGHPUT;
-            isCumulative = false;
-        } else {
-            // When the hint is ctput and there are more than one device, the MULTI logic is used
-            std::string deviceName = "MULTI:";
+        } else if (validDevices.size() > 1) {
+            _loadContext[ACTUALDEVICE].isEnabled = false;
+            _autoSContext->_devicePriorities.clear();
+            for (auto& validDevice : validDevices) {
+                _autoSContext->_devicePriorities.push_back(validDevice);
+            }
+            // Total number of devices in CTPUT
+            _nCTPUTDeviceNums = validDevices.size();
+            DeviceInformation cpuDeviceInformation;
+            const auto CPUIter =
+                std::find_if(validDevices.begin(), validDevices.end(), [=](const DeviceInformation& d) -> bool {
+                    return d.deviceName.find("CPU") != std::string::npos;
+                });
+            if (CPUIter != validDevices.end()) {
+                if (CPUIter->config.find(ov::affinity.name()) == CPUIter->config.end()) {
```

Comment:
insert function will already check the existence of key right? why still need the if here

Reply:
Check here to see if the user has set affinity, if not set, it will be set to CORE by default.

Reply:
I understand, but I mean the insert function will not update the affinity key if already existed.

CPUIter->config.insert(
                        {ov::affinity.name(), ov::affinity(ov::Affinity::CORE).second.as<std::string>()});
// insert the affinity if not in config, else(if this key already in config), do nothing.

Reply:
Updated